### PR TITLE
Unifying 2D and 3D for primitive types

### PR DESCRIPTION
**Pull request by utilForever**
This revision includes:
- Unifying 2D and 3D for primitive types (#92)
  - Transform
  - Surface
  - Collider
  - Plane
  - Box
  - BVH
  - SurfaceSet
  - ColliderSet
  - ImplicitSurface
  - SurfaceToImplicit
  - ImplicitSurfaceSet
  - RigidBodyCollider
  - Sphere
  - ListQueryEngine

### CODE REVIEW

File: Includes/Core/Geometry/BoundingBox.hpp [hunk 1]
```diff
@@ -47,9 +47,6 @@ class BoundingBox
 {
  public:
     static_assert(N > 0, "Dimension should be greater than 0");
-    static_assert(
-        std::is_floating_point<T>::value,
```
Comment (Snowapril) [inline Includes/Core/Geometry/BoundingBox.hpp]: Could I ask you why floating point type assertion was deleted? If given template type N is integer type, below method doesn't work properly. 
```c++
template <typename T, size_t N>
T BoundingBox<T, N>::DiagonalLength() const
{
    return VectorType(upperCorner - lowerCorner).Length();
}
```

Comment (utilForever) [reply Includes/Core/Geometry/BoundingBox.hpp]: Oh, I'll fix it tonight. Thanks. 🕊️

Comment (utilForever) [reply Includes/Core/Geometry/BoundingBox.hpp]: I restored 'static_assert'. 😄

**Review by Snowapril — APPROVED** [review]
It is LGTM now 🥳

**Review by DveloperY0115 — CHANGES_REQUESTED** [review]
Looks cool. Entire geometries are now derived from Surface, thus codes become much easier to read. LGTM now, thanks for your work!

**Review by DveloperY0115 — APPROVED** [review]

**Pull request merged**